apps/dpnk/models/invoice.py

```diff
@@ -341,37 +342,48 @@ def create_and_send_invoice_files(sender, instance, created, **kwargs):
     if created:
         instance.add_payments()
 
-    if not instance.invoice_pdf or not instance.invoice_xml:
-        invoice_data = invoice_gen.generate_invoice(instance)
-        instance.total_amount = invoice_data.price_tax
-        filename = slugify(
-            "invoice_%s_%s_%s_%s"
-            % (
-                instance.sequence_number,
-                instance.company.name[0:40],
-                instance.exposure_date.strftime("%Y-%m-%d"),
-                uuid.uuid4(),
-            ),
-        )
-
-        if not instance.invoice_pdf:
-            temp_pdf = NamedTemporaryFile()
-            invoice_gen.make_invoice_pdf(temp_pdf, invoice_data)
-            instance.invoice_pdf.save("%s.pdf" % filename, File(temp_pdf), save=False)
-
-        if not instance.invoice_xml:
-            temp_xml = NamedTemporaryFile()
-            invoice_gen.make_invoice_xml(temp_xml, invoice_data)
-            instance.invoice_xml.save("%s.xml" % filename, File(temp_xml), save=False)
-
-        instance.save()
-
-    if created:
-        from dpnk.tasks import send_new_invoice_notification
+    if instance.created < settings.FAKTUROID["date_from_create_invoices"]:
+        if not instance.invoice_pdf or not instance.invoice_xml:
+            invoice_data = invoice_gen.generate_invoice(instance)
+            instance.total_amount = invoice_data.price_tax
+            filename = slugify(
+                "invoice_%s_%s_%s_%s"
+                % (
+                    instance.sequence_number,
+                    instance.company.name[0:40],
+                    instance.exposure_date.strftime("%Y-%m-%d"),
+                    uuid.uuid4(),
+                ),
+            )
 
-        send_new_invoice_notification.delay(
-            pks=[instance.pk], campaign_slug=instance.campaign.slug
-        )
+            if not instance.invoice_pdf:
+                temp_pdf = NamedTemporaryFile()
+                invoice_gen.make_invoice_pdf(temp_pdf, invoice_data)
+                instance.invoice_pdf.save(
+                    "%s.pdf" % filename,
+                    File(temp_pdf),
+                    save=False,
+                )
+
+            if not instance.invoice_xml:
+                temp_xml = NamedTemporaryFile()
+                invoice_gen.make_invoice_xml(temp_xml, invoice_data)
+                instance.invoice_xml.save(
+                    "%s.xml" % filename,
+                    File(temp_xml),
+                    save=False,
+                )
+
+            instance.save()
+
+        if created:
+            from dpnk.tasks import send_new_invoice_notification
+
+            send_new_invoice_notification.delay(
+                pks=[instance.pk], campaign_slug=instance.campaign.slug
+            )
+    else:
```

Comment:
Tady je else.

Reply:
Áno máš pravdu `else` vetva tu nemá byť, už som to opravil.